### DISCUSSION TITLE
Bulk Edit Internal Data

### DISCUSSION THREAD

**Discussion by tomferez** [Budibase/budibase#1826] [Help] [answered]
Hi guys,

I'm delving into Budibase and would like to tell you that the app looks really promising!

Is there any way to bulk edit data tables? I can only select one row to edit it. This is an important feature as if I want to migrate from our database into Budibase's own, I need to rebuild relationships. Bulk editing would be necessary, or a script.

Any shared ideas?

Thanks,
Tomas

**Comment by mjashanks** [comment, marked as answer]
Hey Tomas.

There is no way to bulk edit tables via the builder.

This would be possible with a script, however, our APIs are not documented at the moment. We are planning to document these better over the next few months.

If you really want to go there, you can work out the correct calls by viewing what network calls your app does. The process would be like:
1. Pull back all rows in your table, foreach row...
2. Find the correct relationship, using the `POST /api/search/<table id>/rows` endpoint. Have a look at my instructions [here](https://github.com/Budibase/budibase/discussions/1821#discussioncomment-914232) on how to run a filter on a table. This uses the endpoint that I mentioned - just run your app and open `dev tools > network tab`
3. Use the `POST /api/<table id>/rows` endpoint to update the relationship on your record. Again - use your app & dev tools to help you.

Hope this helps :)

**Reply by tomferez** [reply]
Thanks Michael,

Makes sense. I'll try this out! :)
